src/OnboardingSPA/components/Drawer/DrawerPanel/DesignColors.js

```diff
@@ -1,15 +1,378 @@
-import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { Popover, ColorPicker } from '@wordpress/components';
+
+import { store as nfdOnboardingStore } from '../../../store';
+import { getGlobalStyles } from '../../../utils/api/themes';
+import { useGlobalStylesOutput } from '../../../utils/global-styles/use-global-styles-output';
+
 const DesignColors = () => {
-	return (
-		<div style={{ padding: '0 16px' }}>
-			<h2>{__('Pick Color Palette', 'wp-module-onboarding')}</h2>
-			<p>
-				{__(
-					'If user has opted for custom design, panel will show color palettes to affect preview pane.',
-					'wp-module-onboarding'
+
+	const [isLoaded, setIsLoaded] = useState(false);
+	const [globalStyles, setGlobalStyles] = useState();
+	const [selectedColors, setSelectedColors] = useState();
+	const [showColorPicker, setShowColorPicker] = useState(false);
+	const [isAccordionClosed, setIsAccordionClosed] = useState(true);
+
+	const [primaryColor, setPrimaryColor] = useState();
+	const [tertiaryColor, setTertiaryColor] = useState();
+	const [secondaryColor, setSecondaryColor] = useState();
+	const [backgroundColor, setBackgroundColor] = useState();
+	const [colorPickerCalledBy, setColorPickerCalledBy] = useState('');
+
+	const { storedPreviewSettings, currentData } = useSelect(
+		(select) => {
+			return {
+				storedPreviewSettings:
+					select(nfdOnboardingStore).getPreviewSettings(),
+				currentData:
+					select(nfdOnboardingStore).getCurrentOnboardingData(),
+			};
+		},
+		[]
+	);
+
+	const { updatePreviewSettings, setCurrentOnboardingData } =
+		useDispatch(nfdOnboardingStore);
+
+	const colorPalettes = {
+		'calm': [
+			'#C7DBFF',
+			'#E6EBEE',
+			'#1A4733',
+		],
```

Comment:
```suggestion
	const colorPalettes = {
		'calm': {
			'primary': '#C7DBFF',
			'secondary': '#E6EBEE',
			'tertiary': '#1A4733',
		},
```
I think it is Ok to duplicate `primary, secondary, tertiary` as long as we can eliminate confusing indexes for selection. Also should this be coming via an API from the backend based on the active theme?